app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/survey/SurveyDetailScreen.kt

```diff
@@ -69,19 +72,30 @@ fun SurveyDetailScreen(
 ) {
     val currentPage by viewModel.currentPage.collectAsState()
     val surveyQuestions by viewModel.surveyQuestions.collectAsState()
+    val shouldShowLoading by viewModel.shouldShowLoading.collectAsState()
+    val error by viewModel.error.collectAsState()
     var showConfirmDialog by remember { mutableStateOf(false) }
     val startSurveyDescription = stringResource(id = R.string.survey_detail_start_survey)
     val placeholderPainter = rememberAsyncImagePainter(model = survey.coverImagePlaceholderUrl)
     val pagerState = rememberPagerState()
     val scope = rememberCoroutineScope()
 
-    LaunchedEffect(pagerState) {
+    BackHandler {
+        if (currentPage > 0) showConfirmDialog = true
+        else navigator.popBackStack()
```

Comment:
Seems like the dialog is being shown onBackPressed even when I'm looking at one of the questions 🤔

Reply:
<img width="733" alt="Screen Shot 2022-09-08 at 3 03 14 PM" src="https://user-images.githubusercontent.com/32578035/189068869-1b1998fa-7fb2-4b10-b895-2426f90de8d8.png">

I am not sure too. I just implemented the way the documentation says. It says `horizontal swipe` 😅 

> Seems like the dialog is being shown onBackPressed even when I'm looking at one of the questions 🤔

I am not sure what you mean by this. Can you please elaborate?

Reply:
> I just implemented the way the documentation says. It says horizontal swipe 😅

I didn't see this, so yeah no issue here 🤣 

> I am not sure what you mean by this. Can you please elaborate?

Okay I see my misunderstanding. The dialog is supposed to show when viewing questions and not show if viewing the start page. My bad 🙈